File: backend-node-portifolio/src/api/services/interfaces/IServices.ts
export interface IServices<T>{
    Save(entity: T): Promise<string>;
    GetAll(): Promise<T[]>;
    GetbyId(id: number): Promise<T>
    Delete(entity: T): Promise<string>;
    DeleteById(Id: number): Promise<string>;
    Update(entity: T): Promise<string>;
}

File: backend/PostService/src/PostService.Application/Interfaces/IServices.cs
using System.Linq.Expressions;
using System.Linq.Expressions;
namespace PostService.Application.Interfaces
{
    public interface IServices<T>
    {
        Task Save(T entity);
        Task Update(T entity);
        Task Delete(T entity);
        Task DeleteById(Guid Id);
        Task<T> FindBy(Expression<Func<T,bool>> predicate);
        Task<T> GetById(Guid Id);
        Task<List<T>> List();
        Task<List<T>> List(int page);
    }
}